src/entityRequest.ts

```diff
@@ -2,25 +2,24 @@ import { camelizeKeys } from "humps"
 import { authenticatedRequest } from "./requests"
 import { token } from "./token"
 import {
+  ICommunity,
   ICredentials,
   IEnvironment,
   IFetchExecutor,
   IMHBOListing,
+  IUnparsedCommunity,
   IUnparsedMHBOListing
 } from "./types"
 
 /**
- * Performs a search for mobile homes.
+ * Performs a GET request to mhbo api.
  *
  * @param creds The authentication credential for the API request.
  * @param environment An optional override of the environment to utilize.
  * @param fetchExecutor An instance of the request executor.
  * @returns An array of mobile home results.
  */
-async function requestGet<
-  T extends IMHBOListing,
-  U extends IUnparsedMHBOListing
->(
+async function requestGet<T extends any, U extends any>(
```

Comment:
This function has gotten away from us a bit. I would like the entityRequest grouping of code to be the place where you go to see how API requests are made and converted from JSON into their typed responses.  The requestGet function should make the request and then hand the json off to a parsing method depending on what type should be returned.

Something like:
```
   requestGet(url, creds, environment, fetchExecutor, returnType)
      const parseFunction = getParser(returnType) // Or just place them in map to lookup by key
      return ( json.map(parseFunction))
...
   function parseIMHBOListing(result)
...
   function parseIMobileHome(result)
...
   function parseICommunity(result) // this can use parseIMHBOListing for the mobilehomes values
```